examples/src/bin/listbox_model.rs

```diff
@@ -19,6 +19,86 @@ use std::env::args;
 
 use row_data::RowData;
 
+mod model {
+    use super::*;
+    use gio::subclass::ObjectSubclass;
+    use row_data::RowData;
+    mod imp {
+        use super::*;
+        use gio::subclass::ListModelImpl;
+        use glib::subclass;
+        use glib::subclass::prelude::*;
+        use std::cell::RefCell;
+        #[derive(Debug)]
+        pub struct Model(pub RefCell<Vec<RowData>>);
+
+        // Basic declaration of our type for the GObject type system
+        impl ObjectSubclass for Model {
+            const NAME: &'static str = "Model";
+            type Type = super::Model;
+            type ParentType = glib::Object;
+            type Instance = subclass::simple::InstanceStruct<Self>;
+            type Class = subclass::simple::ClassStruct<Self>;
+
+            glib_object_subclass!();
+
+            fn type_init(type_: &mut subclass::InitializingType<Self>) {
+                type_.add_interface::<gio::ListModel>();
+            }
+
+            // Called once at the very beginning of instantiation
+            fn new() -> Self {
+                Self(RefCell::new(Vec::new()))
+            }
+        }
+
+        impl ObjectImpl for Model {}
+
+        impl ListModelImpl for Model {
+            fn get_item_type(&self, _list_model: &Self::Type) -> glib::Type {
+                RowData::static_type()
+            }
+            fn get_n_items(&self, _list_model: &Self::Type) -> u32 {
+                self.0.borrow().len() as u32
+            }
+            fn get_item(&self, _list_model: &Self::Type, position: u32) -> Option<glib::Object> {
+                self.0
+                    .borrow()
+                    .get(position as usize)
+                    .map(|o| o.clone().upcast::<glib::Object>())
+            }
+        }
+    }
+
+    // Public part of the Model type.
+    glib_wrapper! {
+        pub struct Model(ObjectSubclass<imp::Model>) @implements gio::ListModel;
+    }
+
+    // Constructor for new instances. This simply calls glib::Object::new()
+    impl Model {
+        pub fn new() -> Model {
+            glib::Object::new(Self::static_type(), &[])
+                .expect("Failed to create Model")
+                .downcast()
+                .expect("Created Model is of wrong type")
+        }
+
+        pub fn append(&self, obj: &RowData) {
+            let self_ = imp::Model::from_instance(self);
+            self_.0.borrow_mut().push(obj.clone());
+            let index = self_.0.borrow().len() - 1;
```

Comment:
That will become `usize::MAX` for the first item because of wraparound... not sure if that's intended :)